### ISSUE TITLE
Training - Out of Memory Error without GPU

### ISSUE THREAD

**Issue by jsynnott** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#101]
Hi!
Firstly, thank you for such a brilliant pose detection library - it is fantastic.

When attempting to train with just CPU, I am receiving a memory error - are there any parameters I could adjust to reduce the amount of memory required? I don't mind if it takes a long time to run.

A slightly separate question - the reason why I am interested in training myself is because the pre-trained model does not seem to work well with individuals who are lying prone. (e.g. lying on a sofa). I was going to experiment with training the model with all training images rotated 90* CW and CCW to see if this helps. Can you think of a better way to do this that I might be missing?

Thanks very much!

**Comment by Daniil-Osokin** [comment]
Hi! I would suggest to decrease the batch size. It is ok to reduce actual batch size, but save amount of effective batch size (after which gradients are propagated). It can be done using `--batches-per-iter` parameter, so set batch size which fits to the memory (RAM) and add corresponding number of iterations before updating the gradients, i.e. to have effective batch size 80, you may pass the following arguments to the training script: `--batch-size 20 --batches-per-iter 4`. I have observed, that actual batch size of 8 elements takes ~6 GB memory when training on CPU, so adjust your number accordingly.

And for detecting poses under different angles, you may try just to allow [more rotation augmentation](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/datasets/transformations.py#L89) during training. However if your target is lying people I suggest to use dataset where people in such poses are captured more often, like [MPII Human Pose Dataset](http://human-pose.mpi-inf.mpg.de/#download).

**Comment by jsynnott** [comment]
Hi Daniil,
Thank you so much for that very quick and detailed response - that has been an absolutely fantastic help. Very much appreciated!
